### PR TITLE
Test skeleton for rwx volumes

### PR DESCRIPTION
**Pull request by khushboo-rancher**
https://github.com/longhorn/longhorn/issues/1606

I've added a new file test_rwx.py and added a few test skeletons for rwx volumes.

### CODE REVIEW

File: manager/integration/tests/test_rwx.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    3. Wait for longhorn-nfs-provisioner deployment to come up healthy.
+    4. Verify the volume named as the PV attached to longhorn-nfs-provisioner
+       `volume.ready == True`
+    5. Create a StatefulSet of 1 pod with VolumeClaimTemplate and
+       longhorn-nfs class in default namespace.
+    6. Wait for StatefulSet to come up healthy.
+    7. Create another statefulset with same pvc which got created with first
+       statefulset.
+    8. Wait for statefulset to come up healthy.
+    9. Start writing 800 MB data in first stateful set and start writing 500 MB
```
Comment (joshimoo) [inline manager/integration/tests/test_rwx.py:87]: The data should be written to two different files.

File: manager/integration/tests/test_rwx.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+    3. Wait for longhorn-nfs-provisioner deployment to come up healthy.
+    4. Verify the volume named as the PV attached to longhorn-nfs-provisioner
+       `volume.ready == True`
+    5. Create a StatefulSet of 2 pods with VolumeClaimTemplate and
+       longhorn-nfs class in default namespace.
+    6. Wait for both pods to come up running.
+    7. Verify two folders get created with PV names in longhorn-nfs-provisioner
+       volume.
+    8. Write data in both pods and compute md5sum.
```
Comment (joshimoo) [inline manager/integration/tests/test_rwx.py:29]: How do we do this?
Maybe map the export folder from the nfs provisioner?

File: manager/integration/tests/test_rwx.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+       longhorn/longhorn/master/examples/rwx/02-longhorn-nfs-provisioner.yaml
+
+    3. Wait for longhorn-nfs-provisioner deployment to come up healthy.
+    4. Verify the volume named as the PV attached to longhorn-nfs-provisioner
+       `volume.ready == True`
+    5. Create a StatefulSet of 2 pods with VolumeClaimTemplate and
+       longhorn-nfs class in default namespace.
+    6. Wait for StatefulSet pods to come up healthy.
+    7. Write data and compute md5sum in both pods
+    8. Force delete pods.
```
Comment (joshimoo) [inline manager/integration/tests/test_rwx.py:116]: What does force delete refer to in this case?

File: manager/integration/tests/test_rwx.py [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+    3. Wait for longhorn-nfs-provisioner deployment to come up healthy.
+    4. Verify the volume named as the PV attached to longhorn-nfs-provisioner
+       `volume.ready == True`
+    5. Create a StatefulSet of 1 pod with VolumeClaimTemplate and
+       longhorn-nfs class in default namespace.
+    6. Wait for StatefulSet to come up healthy.
+    7. Write data and compute md5sum.
+    8. Shutdown the node where longhorn-nfs-provisioner is running.
+    9. Force delete the deployment in unknown state
```
Comment (joshimoo) [inline manager/integration/tests/test_rwx.py:146]: Why force delete the deployment, you should be able to add a liveness probe which would restart the pod.
That should be sufficient to be able to use the newly brought up nfs-provisioner.

**Comment by joshimoo** [comment]
@khushboo-rancher we should probably add a test with a regular scaled deployment.
If the pods come up on different nodes while they refer to the same PVC that should be sufficient evidence.

File: manager/integration/tests/test_rwx.py (hunk 2 above)
Comment (khushboo-rancher) [reply manager/integration/tests/test_rwx.py:29]: Having pods name and their data path, we can write data to pods.

File: manager/integration/tests/test_rwx.py (hunk 1 above)
Comment (khushboo-rancher) [reply manager/integration/tests/test_rwx.py]: Done

File: manager/integration/tests/test_rwx.py (hunk 3 above)
Comment (khushboo-rancher) [reply manager/integration/tests/test_rwx.py]: Actually force delete is not required, a delete will do. I wrote it to make sure this delete pod in any case.

File: manager/integration/tests/test_rwx.py (hunk 4 above)
Comment (khushboo-rancher) [reply manager/integration/tests/test_rwx.py]: Make sense. Modified it.

**Comment by khushboo-rancher** [comment]
@joshimoo Added a test case with a regular scaled deployment

File: manager/integration/tests/test_rwx.py [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+    11. Write more data to it and compute md5sum.
+    12. Check the data md5sum in longhorn-nfs-provisioner volume
+    """
+    pass
+
+
+@pytest.mark.skip(reason="TODO")
+def test_rwx_deployment_with_multi_pods():
+    """
+    Test moving of longhorn-nfs-provisioner pod from one node to another.
```
Comment (joshimoo) [inline manager/integration/tests/test_rwx.py:159]: Title refers to the previous test.

**Pull request merged**